qiskit/aqua/operators/state_fns/dict_state_fn.py

```diff
@@ -34,7 +34,7 @@ class DictStateFn(StateFn):
 
     # TODO allow normalization somehow?
     def __init__(self,
-                 primitive: Union[str, dict, Result] = None,
+                 primitive: Optional[Union[str, dict, Result]] = None,
```

Comment:
Unless removing the = None for the primitive somehow breaks things I think None is an invalid value for the primitives so I think I would have chosen to delete the `= None` from it rather than add Optional

Reply:
Yes `DictStateFn.__init__` errors when `DictStateFn()` is called because the primitive is not valid.

Reply:
Now, `=None` is necessary for this design pattern.
https://github.com/Qiskit/qiskit-aqua/blob/master/qiskit/aqua/operators/state_fns/state_fn.py#L78-L80

I try to change this.

Reply:
It take many time to remove this None. Could you accept this `None`once. Then, we will fix this in another PR.
This design is not good (1. None hack, 2. Single Responsibility principle). But the range of influence is very large.

Reply:
Sure, if its complicated to sort things out at present then I am ok keeping the `= None` Perhaps though we should not add the `Optional` - i.e. leave it like it was. Adding the Optional seems to strengthen the case that None is a valid value to pass when truly it is not.

Reply:
Ok, I will remove `Optional`.